### PR TITLE
[SYCLomatic] Fix the incorrent migration of cudaEventRecord() when its argument is unary operator "*"

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 1]
```diff
@@ -7222,6 +7222,11 @@ void EventAPICallRule::handleEventRecordWithProfilingEnabled(
   auto IndentLoc = CE->getBeginLoc();
   auto &SM = DpctGlobalInfo::getSourceManager();
 
+  if (const auto *UO = dyn_cast<UnaryOperator>(StreamArg->IgnoreImpCasts())) {
```
Comment (zhiweij1) [inline clang/lib/DPCT/ASTTraversal.cpp]: Is it better to compare the precedence of `*` and `->`?

Comment (tomflinda) [reply clang/lib/DPCT/ASTTraversal.cpp]: Yes, updated.

File: clang/test/dpct/cuda-event-api-enable-profiling.cu [hunk 2]
```diff
@@ -347,3 +347,24 @@ void foo2(Node *n) {
...
+
+// CHECK:static int cuda_stream_decode_ioinstruction(foo_stream_t *ios) {
+// CHECK-NEXT:  foo__io_stream_t *cios = (foo__io_stream_t *)ios;
+// CHECK-NEXT:  dpct::queue_ptr *stream = 0;
+// CHECK-NEXT:  *cios->end_events[ios->pos_wp % ios->count] = (*stream)->ext_oneapi_submit_barrier();
+// CHECK-NEXT:}
+static int cuda_stream_decode_ioinstruction(foo_stream_t *ios) {
+  foo__io_stream_t *cios = (foo__io_stream_t *)ios;
+  cudaStream_t *stream = 0;
+  cudaEventRecord(cios->end_events[ios->pos_wp % ios->count], *stream);
```
Comment (zhimingwang36) [inline clang/test/dpct/cuda-event-api-enable-profiling.cu:368]: let's do same for the 1st arguments

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Review by zhiweij1 — APPROVED** [review]

**Pull request merged**